Content/SnippetResolver.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\HeadlessBundle\Content;
+
+use Sulu\Component\Content\Compat\Structure\SnippetBridge;
+use Sulu\Component\Content\Mapper\ContentMapperInterface;
+
+class SnippetResolver implements SnippetResolverInterface
+{
+    /**
+     * @var ContentMapperInterface
+     */
+    private $contentMapper;
+
+    /**
+     * @var StructureResolverInterface
+     */
+    private $structureResolver;
+
+    public function __construct(ContentMapperInterface $contentMapper, StructureResolverInterface $structureResolver)
+    {
+        $this->contentMapper = $contentMapper;
+        $this->structureResolver = $structureResolver;
+    }
+
+    /**
+     * @return mixed[]
+     */
+    public function resolve(
+        string $uuid,
```

Comment:
i think the other resolvers do not load the data by themselves but get the loaded data passed to the `resolve` function, eg the `StructureResolver` receives a `StructureBridge $structure` here

Reply:
i think we should pass a `SnippetBridge $snippet` here for consistency because of this

Reply:
Yes you're right, but then the functionality regarding loading the snippet in the shadow locale, if the original locale is not available, will be duplicated in the `SingleSnippetSelectionResolver` and the `SnippetSelectionResolver` ... But on the other hand, then this service would really just be a serializer

Reply:
i think it would be okay for me to duplicate the fallback to the shadow locale in the `SingleSnippetSelectionResolver` and `SnippetSelectionResolver`. i guess this is only required when a snippet is resolved as part of a page/article, right?

Reply:
implementing this as `SnippetSerializer` in the `Serializer` namespace would feel better for me too. but dont have a too strong opinion 🙂

Reply:
Then we don't even need a service for that, because we can just use the `StructureResolver` directly

Reply:
thats right, sorry 🙈  can you think about a case where we need a shadow locale outside of the `SingleSnippetSelectionResolver`/`SnippetSelectionResolver`? @sulu/core-team

Reply:
Okay seem slike there is no shadow handling needed in the controller so it sounds good for me if the controller then just does.


```php
$snippet = $defaultSnippetManager->load($defaultSnippet);

$structureResolver->resolve($snippet, ...);
```

And the  call just `SingleSnippetSelection` calls like [`SingleMediaSelection`](https://github.com/sulu/SuluHeadlessBundle/blob/0.x/Content/ContentTypeResolver/SingleMediaSelectionResolver.php) the `SnippetSelection` so we have the shadow handling only at this case.